File: 1779.find-nearest-point-that-has-the-same-x-or-y-coordinate/solution.py
class Solution:
    def nearestValidPoint(self, x: int, y: int, points: List[List[int]]) -> int:
        nearestDistance = float('inf')
        nearestIdx = -1

        for i, point in enumerate(points):
            if (point[0] != x) and (point[1] != y):
                continue

            distance = sqrt(abs(point[0] - x) ** 2
                            + abs(point[1] - y) ** 2)

            if (nearestDistance <= distance):
                continue

            nearestDistance = distance
            nearestIdx = i

        return nearestIdx
            

File: 1779.find-nearest-point-that-has-the-same-x-or-y-coordinate/solution.kt
class Solution {
    fun nearestValidPoint(x: Int, y: Int, points: Array<IntArray>): Int {
        var nearestDistance = 999999999.0
        var nearestIdx = -1

        for (i in 0..points.size - 1) {
            val point = points[i]
            if ((point[0] != x) && (point[1] != y)) {
                continue
            }

            val distance = Math.sqrt(
                Math.pow(Math.abs(point[0] - x).toDouble(), 2.0) +
                    Math.pow(Math.abs(point[1] - y).toDouble(), 2.0)
            )

            if (nearestDistance <= distance) {
                continue
            }

            nearestDistance = distance
            nearestIdx = i
        }
        return nearestIdx
    }
}
